util-core/src/main/scala/com/twitter/io/ByteReader.scala

```diff
@@ -384,6 +398,41 @@ private class ByteReaderImpl(buf: Buf) extends ByteReader {
     ret
   }
 
+  private val unsignedLongMaxValue: BigInt = BigInt("18446744073709551615")
+
+  def readUnsignedLongBE(): BigInt = {
```

Comment:
I fixed unnecessary allocations.

> also, why do we need to & it with the unsignedLongMaxValue?

Frankly, I saw that in existing functions, for instance, _readUnsignedIntLE,_ I guess they're used just in case. Maybe to reset other bits to 0.